scheduler/util.go

```diff
@@ -289,15 +286,75 @@ func readyNodesInDCs(state State, dcs []string) ([]*structs.Node, map[string]int
 		if !node.Ready() {
 			continue
 		}
-		if _, ok := dcMap[node.Datacenter]; !ok {
+
+		match := matchDcs(dcs, node)
+
+		if !match {
 			continue
 		}
+
 		out = append(out, node)
 		dcMap[node.Datacenter]++
 	}
 	return out, dcMap, nil
 }
 
+func matchDcs(dcs []string, node *structs.Node) bool {
+	for _, dc := range dcs {
+		match := globMatch(dc, node.Datacenter)
+
+		if match {
+			return true
+		}
+	}
+
+	return false
+}
+
+// globMatch is used to match names against simple glob patterns
+func globMatch(pattern, name string) bool {
```

Comment:
Great use of Russ Cox's algorithm! Let's use `github.com/ryanuber/go-glob#Glob` instead as we already use it for matching ACLs and Docker's `extra_labels` parameter.

It doesn't support `?`, but I don't think that's a big loss.

Sorry for not pointing out this library initially!